api/v1beta1/hostedcluster_types.go

```diff
@@ -737,6 +737,72 @@ type KubevirtPlatformSpec struct {
 	// the same cluster and namespace as the Hosted Control Plane.
 	// +optional
 	Credentials *KubevirtPlatformCredentials `json:"credentials,omitempty"`
+
+	// StorageDriver defines how the KubeVirt CSI driver exposes StorageClasses on
+	// the infra cluster (hosting the VMs) to the guest cluster.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	// +immutable
+	StorageDriver *KubevirtStorageDriverSpec `json:"storageDriver,omitempty"`
+}
+
+// KubevirtStorageDriverConfigType defines how the kubevirt storage driver is configured.
+//
+// +kubebuilder:validation:Enum=None;Default;Manual
+type KubevirtStorageDriverConfigType string
+
+const (
+	// NoneKubevirtStorageDriverConfigType means no kubevirt storage driver is used
+	NoneKubevirtStorageDriverConfigType KubevirtStorageDriverConfigType = "None"
+
+	// DefaultKubevirtStorageDriverConfigType means the kubevirt storage driver maps to the
+	// underlying infra cluster's default storageclass
+	DefaultKubevirtStorageDriverConfigType KubevirtStorageDriverConfigType = "Default"
+
+	// ManualKubevirtStorageDriverConfigType means the kubevirt storage driver mapping is
+	// explicitly defined.
+	ManualKubevirtStorageDriverConfigType KubevirtStorageDriverConfigType = "Manual"
+)
+
+type KubevirtStorageDriverSpec struct {
+	// Type represents the type of kubevirt csi driver configuration to use
+	//
+	// +unionDiscriminator
+	// +immutable
+	// +kubebuilder:default=Default
+	// +optional
+	Type KubevirtStorageDriverConfigType `json:"type,omitempty"`
+
+	// Manual is used to explicilty define how the infra storageclasses are
+	// mapped to guest storageclasses
+	//
+	// +immutable
+	// +optional
+	Manual *KubevirtManualStorageDriverConfig `json:"manual,omitempty"`
+}
+
+type KubevirtManualStorageDriverConfig struct {
+	// StorageClassMapping maps StorageClasses on the infra cluster hosting
+	// the KubeVirt VMs to StorageClasses that are made available within the
+	// Guest Cluster.
+	//
+	// NOTE: It is possible that not all capablities of an infra cluster's
+	// storageclass will be present for the corresponding guest clusters storageclass.
+	//
+	// +optional
+	// +immutable
```

Comment:
For my info, do you mind sharing how immutability is actually enforced here? I don't anything at the code/schema level in this pr that makes it possible. But I'm sure I'm missing something

Reply:
fixed... Yeah i think something changed in the way the crd schema is built recently. I added a CEL validation to enforce all this immutability stuff, and it seems to work.